lib/storage/environment.go

```diff
@@ -107,9 +109,52 @@ func (r *EnvironmentV1) GetKeyValues() map[string]string {
 
 // CheckAndSetDefaults validates this resource and sets defaults
 func (r *EnvironmentV1) CheckAndSetDefaults() error {
+	var errors []error
+	if err := r.checkHTTPProxy(); err != nil {
+		errors = append(errors, err)
+	}
+	if err := r.checkHTTPSProxy(); err != nil {
+		errors = append(errors, err)
+	}
+	return trace.NewAggregate(errors...)
+}
+
+// checkHTTPProxy verifies HTTP_PROXY (if present) is a valid proxy URL.
+func (r *EnvironmentV1) checkHTTPProxy() error {
+	httpProxy := r.getVariable(constants.HTTPProxyEnvVar)
+	if httpProxy == "" {
+		return nil
+	}
+	if _, err := utils.ParseProxy(httpProxy); err != nil {
+		return trace.Wrap(err, "failed to parse HTTP_PROXY")
+	}
 	return nil
 }
 
+// checkHTTPSProxy verifies HTTPS_PROXY (if present) is a valid proxy URL.
+func (r *EnvironmentV1) checkHTTPSProxy() error {
```

Comment:
Nit: I think checkHTTPProxy / checkHTTPSProxy are largely the same function now. They're not long, but usually I get called out for having anything that looks like duplicate code.